File: medical-ai-app/frontend/src/types/dashboard(1).ts
export interface DashboardSummary {
  total_cases: number;
  today_cases: number;
  average_accuracy: number;
  active_modules: number;
  search_matches: Array<{ id: number; case_code: string; module_type: string }>;
  performance: {
    dice: number;
    accuracy: number;
    confidence: number;
  };
  quick_tips: string[];
  notifications: Array<{
    id: number;
    title: string;
    description: string;
    level: string;
    created_at: string;
  }>;
}

export interface RecentCase {
  id: number;
  case_code: string;
  module_type: string;
  created_at: string;
  predicted_label?: string | null;
  status: string;
  confidence?: number | null;
}

File: medical-ai-app/backend/app/schemas/dashboard(1).py
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class DashboardSummary(BaseModel):
    total_cases: int
    today_cases: int
    average_accuracy: float
    active_modules: int
    search_matches: list[dict[str, Any]] = []
    performance: dict[str, float]
    quick_tips: list[str]
    notifications: list[dict[str, Any]]


class RecentCaseItem(BaseModel):
    id: int
    case_code: str
    module_type: str
    created_at: datetime
    predicted_label: str | None = None
    status: str
    confidence: float | None = None


class RecentCasesResponse(BaseModel):
    items: list[RecentCaseItem]
